src/test/java/org/jooq/lambda/SeqTest.java

```diff
@@ -1543,6 +1543,22 @@ public void testIterate() {
     }
 
     @Test
+    public void testIterateWhilePresent() {
+        assertEquals(
+              asList(),
+              Seq.iterateWhilePresent(null, i -> Optional.empty()).toList()
+        );
```

Comment:
I find this surprising. This should be equal to `asList((Object) null)`. Illustration:

https://github.com/jOOQ/jOOL/commit/0ba503536cefa8c30e0697d2c0c4ce2e8e013f7a